themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/signingin-page/SigningInPage.tsx

```diff
@@ -0,0 +1,183 @@
+            isTotpConfigured: features.isTotpConfigured,
+        }
+        this.setLastPwdUpdate();
+    }
+
+    private setLastPwdUpdate(): void {
+        AccountServiceClient.Instance.doGet("/credentials/password")
+            .then((response: AxiosResponse<PasswordDetails>) => {
+                if (response.data.lastUpdate) {
+                    const lastUpdate: number = response.data.lastUpdate;
+                    this.setState({lastPasswordUpdate: lastUpdate});
+                }
+            });
+    }
+    
+    private handleTwoFactorSwitch = () => {
+        if (this.state.twoFactorEnabled) {
+            this.setState({twoFactorEnabled: false, twoFactorEnabledText: Msg.localize('twoFactorDisabled')})
+        } else {
+            this.setState({twoFactorEnabled: true, twoFactorEnabledText:  Msg.localize('twoFactorEnabled')})
+        }
+    }
+
+    private handleRemoveTOTP = () => {
+      AccountServiceClient.Instance.doDelete("/totp/remove")
+        .then(() => {
+            this.setState({isTotpConfigured: false});
+            ContentAlert.success('successTotpRemovedMessage');
+        });
+    }
+
+    public render(): React.ReactNode {
+        let lastPwdUpdate: string = Msg.localize('unknown');
+        if (this.state.lastPasswordUpdate) {
+            lastPwdUpdate = moment(this.state.lastPasswordUpdate).format('LLL');
+        }
+
+        return (
+            <ContentPage title="signingIn" 
+                     introMessage="signingInSubMessage">
+                <Stack gutter='md'>
+                    <StackItem isFilled>
+                        <Title headingLevel={TitleLevel.h2} size='2xl'>
+                        <strong><Msg msgKey='password'/></strong>
+                        </Title>
+                        <DataList aria-label='foo'>
+                            <DataListItemRow>
+                                <DataListItemCells
+                                    dataListCells={[
+                                        <DataListCell key='password'><Msg msgKey='password'/></DataListCell>,
+                                        <DataListCell key='lastPwdUpdate'><strong><Msg msgKey='lastUpdate'/>: </strong>{lastPwdUpdate}</DataListCell>,
+                                        <DataListCell key='spacer'/>
+                                    ]}/>
+                                <DataListAction aria-labelledby='foo' aria-label='foo action' id='setPasswordAction'>
+                                    <Button variant='primary'onClick={()=> this.updatePassword.execute()}><Msg msgKey='update'/></Button>
+                                </DataListAction>
+                            </DataListItemRow>
+                        </DataList>
+                    </StackItem>
+                    <StackItem isFilled>
+                        <Title headingLevel={TitleLevel.h2} size='2xl'>
+                        <strong><Msg msgKey='twoFactorAuth'/></strong>
```

Comment:
Shouldn't we move the two factor authentication to the same box with password? This way we could make the impression that it could be used with password less credentials as well which AFAIK is not possible.

Reply:
That's a question for the keycloak-dev list.  This comes from Stian's mockup here: https://docs.google.com/document/d/1Cmon4ODQyd-wvcEbHNJrnZAC3VAxDLM1DB6mjOcjdWk/edit